metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala

```diff
@@ -13,17 +19,21 @@ import org.eclipse.lsp4j.SemanticTokenTypes
  *  Provides semantic tokens of file
  *  according to the LSP specification.
  */
-abstract class SemTokenProvider {
-  type Sym
-  type Pos
-
-  val capableTypes = SemanticTokens.TokenTypes
-  val capableModifiers = SemanticTokens.TokenModifiers
-  // Alias for long notation
-  val getTypeId: Map[String, Int] = capableTypes.zipWithIndex.toMap
-  val getModifierId: Map[String, Int] = capableModifiers.zipWithIndex.toMap
-  def getTokens(): Tokens
-  def nodes: List[NodeInfo]
+class SemanticTokensProvider(
+    compiler: PresentationCompiler,
```

Comment:
```suggestion
    isScala3Version: Boolean,
```
?

These can also be paramters to `provide` method and we could make it an object.